climakitae/threshold_tools.py

```diff
@@ -884,3 +889,523 @@ def return_period(ams):
+        self.selections.variable = self.variable2
+
+    @param.depends("area_subset2", watch=True)
+    def _update_cached_area(self):
+        """
+        Makes the dropdown options for 'cached area' reflect the type of area subsetting
+        selected in 'area_subset' (currently state, county, or watershed boundaries).
+        """
+        if self.area_subset2 == "CA counties":
+            # Get full list of counties
+            self.param["cached_area2"].objects = list(
+                self.location._geography_choose[self.area_subset2].keys()
+            )
+            self.cached_area2 = "Sacramento County"
+        elif self.area_subset2 == "states":
+            # Only CA for now?
+            self.param["cached_area2"].objects = ["CA"]
+            self.cached_area2 = "CA"
+
+    @param.depends("area_subset2","cached_area2",watch=True)
+    def _updated_location(self):
+        """Update locations object to reflect location chosen in panel"""
+        self.location.area_subset = self.area_subset2
+        self.location.cached_area = self.cached_area2
+
+    def transform_data(self):
+        return get_exceedance_count(self.da, 
+            threshold_value = self.threshold_value, 
+            threshold_direction = self.threshold_direction,
+            duration1 = (self.duration1_length, self.duration1_type),
+            period = (self.period_length, self.period_type),
+            groupby = (self.group_length, self.group_type),
+            duration2 = (self.duration2_length, self.duration2_type),
+            smoothing = self.num_timesteps if self.smoothing == "Running mean" else None)
+
+    @param.depends("reload_plot", "reload_data", watch=False)
+    def view(self):
+        try:
+            to_plot = self.transform_data()
+            obj = plot_exceedance_count(to_plot)
+        except Exception as e:
+            # Display any raised Errors (instead of plotting) if any of the 
+            # user specifications are incompatible or not yet implemented.
+            return e
+        return pn.Column(
+            pn.widgets.Button.from_param(self.param.reload_plot, button_type="primary", width=150, height=30),
+            _exceedance_plot_title(to_plot),
+            _exceedance_plot_subtitle(to_plot),
+            obj
+        )
+        return obj
+
+    @param.depends("smoothing")
+    def smoothing_card(self):
+        """A reactive panel card used by _exceedance_visualize that only 
+        displays the num_timesteps option if smoothing is selected."""
+        if self.smoothing != "None":
+            smooth_row = pn.Row(
+                self.param.smoothing,
+                self.param.num_timesteps, 
+                width=375
+            )
+        else:
+            smooth_row = pn.Row(
+                self.param.smoothing,
+                width=375
+            )
+        return pn.Card(smooth_row, title = "Smoothing", collapsible = False)
```

Comment:
The smoothing card looks a little strange a lonely hanging out on its own below the threshold selections panel. I would suggest incorporating it into the threshold events selection Card.